### PR TITLE
fix(762): migrate reputation route lookupAgent to shared helper + negative-on-timeout cache

### PR DESCRIPTION
**Pull request by whoabuddy**
## Summary

Defensive cleanup for the reputation sub-route. Migrates its local KV-only `lookupAgent` to the shared D1-capable helper (closing the gap #788 missed), and adds a 60s negative cache on Hiro `TimeoutError` to break the polling-storm amplification observed at #788 T+4h.

## Why

Observed at PR #788 T+4h cost gate (`2026-05-13T02:48:17Z`): 12 `reputation.fetch_error` / `reputation.summary_fetch_error` rows over 7min, 6 distinct cf-rays, all `TimeoutError: The operation was aborted due to timeout` against `/api/identity/<bc1q…>/reputation` for agent-id 42. No negative cache → each poll cold-isolates → sustained Hiro pressure.

The BTC→STX side wasn't the problem (the local `lookupAgent` resolved fine), but it's still a KV-only path that #788 should have migrated. Cleanup-in-flight.

## Changes

1. **Migrate local `lookupAgent` in `app/api/identity/[address]/reputation/route.ts`** to `@/lib/agent-lookup` (D1-capable, threaded `db` per #788 pattern). Both call sites updated. Shared helper signature is `(kv, address, db?)` — no logger parameter (uses `console.error` internally); noted here per spec.
2. **Negative-on-timeout cache in `getReputationSummary`** — 60s TTL on `TimeoutError` / 5xx, via new `setCachedReputationLookupFailed` helper added to `kv-cache.ts`. Mirrors the existing `setCachedBnsLookupFailed` / `setCachedIdentityLookupFailed` pattern (NONE_SENTINEL, 60s TTL per CLAUDE.md "Cache Model for BNS + Identity Lookups").
3. **Symmetric treatment in `getReputationFeedback`** — same structure, same amplification risk.

## Affected components

| Component | Change |
|---|---|
| `app/api/identity/[address]/reputation/route.ts` | Local `lookupAgent` removed; shared helper imported; `db` binding threaded from env |
| `lib/identity/reputation.ts` | `getReputationSummary` + `getReputationFeedback` now write a 60s negative cache on transient failures before rethrowing |
| `lib/identity/kv-cache.ts` | New `setCachedReputationLookupFailed` helper (NONE_SENTINEL, 60s TTL) |
| `lib/__tests__/reputation-negative-cache.test.ts` | 4 new tests covering timeout→cache and cache-hit→no-Hiro paths for both functions |

## Verify

- Targeted vitest: 4/4 new tests pass; full suite 1939/1939 pass, 0 failures.
- Typecheck Δ=0 (133 baseline, 133 now).
- Worker-logs at next T+24h sweep should not show recurring polling-storm pattern for slow-Hiro contracts (will land at #788 T+24h gate @ 2026-05-13T22:48:17Z).

## Risk

Low. Step 1 is a code-dedup refactor (same external behavior for BTC addresses; STX addresses now go D1-first, fail-closed per Phase 4.0d). Steps 2/3 add caching on a failure path; worst-case is a stale 60s negative — bounded and matches existing BNS/identity pattern.

Closes #795.
Refs #762.

### CODE REVIEW

**Review by arc0btc — APPROVED** [review]
Defensive fix for the reputation polling-storm observed at #788 T+4h — migrates the stale KV-only `lookupAgent` to the shared D1-capable helper and adds a 60s negative cache on transient Hiro failures. Clean, well-scoped change.

**What works well:**
- `setCachedReputationLookupFailed` mirrors the existing `setCachedBnsLookupFailed` / `setCachedIdentityLookupFailed` pattern exactly — same NONE_SENTINEL, same 60s TTL, same `kvPut` path. Easy to audit.
- Removing 27 lines of local `lookupAgent` and replacing with the shared helper is the right dedup move. The `(kv, address, db?)` signature upgrade also silently fixes the STX address gap (#788 missed this call site).
- `db = env.DB as D1Database | undefined` is the correct way to thread the optional binding — fail-open to KV-only if DB isn't bound.
- The `Promise.all()` on `clientAddresses.map` was already there and remains parallel — no N+1 regression.
- 4 targeted tests cover both the write-and-rethrow path and the cache-hit-short-circuit path for each function. Good coverage for the added behavior.

**[question] Structured logging lost in shared helper** (`app/api/identity/[address]/reputation/route.ts:103`)
The local `lookupAgent` accepted a `logger` and called `logger.error("reputation.parse_agent_failed", ...)` on parse failures. The shared helper uses `console.error` internally per the PR description. For parse errors on agent records this is a structured-log regression — `console.error` won't produce a `reputation.parse_agent_failed` row in worker analytics. Is this intentional / already tracked somewhere, or worth threading a logger into the shared helper later?

**Code quality notes:**
- The inline comments in `reputation.ts` explaining the negative-cache rationale are good — the "why" (mirrors BNS pattern, breaks polling-storm amplification) isn't obvious from the code alone. Worth keeping.
- `await setCachedReputationLookupFailed(...)` before `throw` adds one KV write latency to the error path (~5-10ms in a CF worker). Totally acceptable for the protection it provides; just noting it so nobody is surprised when error-path traces show a small extra step.

**Performance / Composition / UI/Accessibility**

No React components or frontend code changed — performance, composition, and accessibility rules don't apply to this diff.

Operationally: we poll `/api/identity/.../reputation` through our sensors and have hit exactly this TimeoutError pattern against slow contracts. The 60s negative cache will help on our end too.

File: lib/identity/reputation.ts [hunk 1]
```diff
@@ -63,6 +63,12 @@ export async function getReputationSummary(
       agentId,
       error: String(error),
     });
+    // Write a 60s negative cache before rethrowing so that concurrent and
+    // subsequent polls within the transient-error window hit the cache
+    // instead of piling onto a Hiro endpoint that's already in timeout/5xx.
+    // Mirrors the setCachedBnsLookupFailed / setCachedIdentityLookupFailed
+    // pattern from kv-cache.ts (60s TTL, NONE_SENTINEL value).
+    await setCachedReputationLookupFailed(cacheKey, kv, logger);
     throw error;
```
[the remark this answers is not in the document]
Comment (whoabuddy) [reply lib/identity/reputation.ts]: Substantive — fixed in commit 16cb3e4. Changed both catch blocks to write the negative cache and **return null/empty silently** (matching the BNS/identity `setCachedBnsLookupFailed`/`setCachedIdentityLookupFailed` pattern). The 1st-vs-subsequent-call asymmetry you flagged is gone: both calls now return null. Trade-off accepted: API surface shifts from "500 on upstream failure" → "200 with null reputation during 60s window" — bounded by TTL, consistent across the polling client. Tests + JSDoc updated to reflect the new shape.

File: lib/identity/kv-cache.ts [hunk 2]
```diff
@@ -477,6 +477,31 @@ export function setCachedReputation(
...
+  logger?: Logger
+): Promise<void> {
+  return kvPut(
+    kv,
+    `cache:reputation:${key}`,
+    NONE_SENTINEL,
+    60,
+    "reputation",
+    logger
+  );
```
[the remark this answers is not in the document]
Comment (whoabuddy) [reply lib/identity/kv-cache.ts:513]: Fixed in 16cb3e4 — extracted `REPUTATION_LOOKUP_FAILED_CACHE_TTL = 60` near the other lookup-failed TTLs in `kv-cache.ts`; `setCachedReputationLookupFailed` now references the constant. No more magic 60.

File: lib/__tests__/reputation-negative-cache.test.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  getCachedReputation: vi.fn(),
+  setCachedReputation: vi.fn().mockResolvedValue(undefined),
+  setCachedReputationLookupFailed: vi.fn().mockResolvedValue(undefined),
+}));
+
+// ---- imports ----------------------------------------------------------------
+
+import { getReputationSummary, getReputationFeedback } from "@/lib/identity/reputation";
+import { callReadOnly, parseClarityValue } from "@/lib/identity/stacks-api";
+import { getCachedReputation, setCachedReputationLookupFailed } from "@/lib/identity/kv-cache";
```
[the remark this answers is not in the document]
Comment (whoabuddy) [reply lib/__tests__/reputation-negative-cache.test.ts]: Fixed in 16cb3e4 — removed unused `parseClarityValue` import (and the mock export was reused elsewhere, so kept in the vi.mock block). Also rewrote the file docstring + (a)/(c) test bodies to reflect the new "silent null/empty" behavior.
